### PR TITLE
Moves all resarched weapons to a seperate armory protolathe.

### PR DESCRIPTION
**Pull request by Fresh839**
## About The Pull Request
Moves all weapons unlocked via weapon technologies to a seperate protolathe in the armory.

## Why It's Good For The Game
This allows the warden to control who gets advanced weapons and who doesn't. 
I never liked how you could deny a bloodthirsty security officer a weapon and they would just wait a bit and then print an even better one from the security protolathe, to which even the detective has access to. Having security officers walk around with marksman beam rifles on a greenshift is not fun for anyone. This doesn't remove that possibility, just allows you to blame it on the warden in that case.

This does NOT affect any other equipment besides the weapons. Telescopic Shields etc. are still available from the normal security protolathe. Firing pins are available from both lathes as a QOL.
Ammunition will still be available from the normal security protolathe. Its just that the guns itself can now only be printed in the armory.
This does also NOT affect mech weapons. They are still available from the normal security lathe.
EDIT: Forgot to add, this moves landmines/stun mines to the armory lathe. EDIT2: Nevermind i am stupid, it doesnt.

I added a lathe to all five maps. (Box,Meta,Delta,Pubby and Corg) I tried to keep the changes as minimal as possible but if someone has a problem with it please let me know.

This is my first pr ever, so bear with me if i fucked things up. I tested all the changes though and it seemed to work fine.

## Changelog
:cl:
add: Added a new departement called "armory"
add: Added a new departemental protolathe to the armory departement
add: Added the armory protolathe board to the HOS locker.
add: Added mindshield and test-range firing pins to the armory protolathe
balance: Moved tesla revolver, advanced energy gun, marksman beamrifle, decloner, temp gun, x-ray gun, ion carbine and energy crossbow to said new protolathe
/:cl:

### CODE REVIEW

**Comment by OceanFish1** [comment]
Validhunting rate drops to 0% congratulations

**Comment by PowerfulBacon** [comment]
I don't really like the corg placement

File: code/__DEFINES/research.dm [hunk 1]
```diff
@@ -56,6 +56,7 @@
 #define DEPARTMENTAL_FLAG_ENGINEERING	(1<<4)
 #define DEPARTMENTAL_FLAG_SERVICE		(1<<5)
 #define DEPARTMENTAL_FLAG_ALL			(1<<6)			//NO THIS DOESN'T ALLOW YOU TO PRINT EVERYTHING, IT'S FOR ALL DEPARTMENTS!
+#define DEPARTMENTAL_FLAG_ARMORY		(1<<7)
```
Comment (PowerfulBacon) [inline code/__DEFINES/research.dm]: Armory should be before ALL

**Review by PowerfulBacon — DISMISSED** [review]

File: code/__DEFINES/research.dm (hunk 1 above)
Comment (Fresh839) [reply code/__DEFINES/research.dm]: Aight, let me see if i can change this.

**Comment by ItsTisme** [comment]
besure to give either HoS or warden a spare weapons lathe board. Its standard for there to be a spare in the lockers

**Comment by Fresh839** [comment]
> 
> 
> besure to give either HoS or warden a spare weapons lathe board. Its standard for there to be a spare in the lockers

It is in the changelog

**Comment by Fresh839** [comment]
> 
> 
> I don't really like the corg placement

Hmm i can place it somewhere else if you want me to. Any idea where it would fit better?

**Comment by Isy232** [comment]
> add: Added all firing pins to the armory protolathe

Do you mean you also allowed it to print standard firing pins instead of just mindshield and test range?

If so, thats good, but please tell me.

**Comment by Fresh839** [comment]
> 
> 
> > add: Added all firing pins to the armory protolathe
> 
> Do you mean you also allowed it to print standard firing pins instead of just mindshield and test range?
> 
> If so, thats good, but please tell me.

No, i only added the mindshield and test-range firing pins to them. I couldn't find the standard ones in the weapon_designs.dm. Edited the changelog now. I can try to add them too but i would first have to find them.

**Comment by francinum** [comment]
Adding standard firing pins to the armory lathe will not be approved.
They are intentionally not printable.

Do not include edit tags in the changelog segment.

**Comment by Fresh839** [comment]
> 
> 
> Adding standard firing pins to the armory lathe will not be approved.
> They are intentionally not printable.
> 
> Do not include edit tags in the changelog segment.

Alright, well i would have no idea on how to add them anyways. Sorry for the edit tag then, i didn't know that. Anything else that you don't agree with/needs fixing?

**Review by PowerfulBacon — DISMISSED** [review]
Corgstation is fine, but rip the sec scooter

**Comment by Fresh839** [comment]
> 
> 
> Corgstation is fine, but rip the sec scooter

Oh i did not remove the scooter, i just moved it ontop of the other one. I did the same with the flashers on another map. Thats what i mean't with minimal changes. I hope thats still okay.

**Comment by Fresh839** [comment]
Hope that last commit removed the merge conflicts. (I hope i did not break everything here)

**Review by PowerfulBacon — DISMISSED** [review]

File: _maps/map_files/CorgStation/CorgStation.dmm [hunk 2]
```diff
@@ -48634,6 +48616,41 @@
...
+	dir = 8
+	},
+/obj/machinery/atmospherics/pipe/simple/scrubbers/hidden/layer3{
+	dir = 4
+	},
+/obj/vehicle/ridden/secway,
+/turf/open/floor/plasteel/dark,
+/area/ai_monitored/security/armory)
+=======
+>>>>>>> master
```
Comment (bloons3) [inline _maps/map_files/CorgStation/CorgStation.dmm]: merge conflict, fails map CI

**Comment by Fresh839** [comment]
I won't have access to my computer in the next 10 or so days, i will see if i can fix everything once i come back!

**Comment by BrumalLake** [comment]
Porting [tgstation#56133](https://github.com/tgstation/tgstation/pull/56133) is a much better solution than this band-aid

**Comment by Fresh839** [comment]
> 
> 
> Porting [tgstation#56133](https://github.com/tgstation/tgstation/pull/56133) is a much better solution than this band-aid

I don't really understand what you mean with this. The PR you mentioned completely revamps how security aquires advanced guns. It completely changes the balance and would never pass the public vote in my opinion.

And your PR still does not fix the issue that any officer can aquire those advanced guns. Which questions me if you even read what my PR is supposed to "fix"/change. My PR tries to make the warden be responsible for the advanced weaponry of security. And not like how it is right now, everyone with just basic security access can print particle acceleration rifles or any other really strong gun for example. The TG PR still allows any security officer to craft those advanced guns. Read my PR description please before commenting.

**Comment by BrumalLake** [comment]
> 
> 
> > Porting [tgstation#56133](https://github.com/tgstation/tgstation/pull/56133) is a much better solution than this band-aid
> 
> I don't really understand what you mean with this. The PR you mentioned completely revamps how security aquires advanced guns. It completely changes the balance and would never pass the public vote in my opinion.
> 
> And your PR still does not fix the issue that any officer can aquire those advanced guns. Which questions me if you even read what my PR is supposed to "fix"/change. My PR tries to make the warden be responsible for the advanced weaponry of security. And not like how it is right now, everyone with just basic security access can print particle acceleration rifles or any other really strong gun for example. The TG PR still allows any security officer to craft those advanced guns. Read my PR description please before commenting.

They need a base gun from the armory or cargo to first create the guns, which, I'd like to think, the warden/hos takes care of, as they are the only ones who can grant guns from the armory as your pr is claiming and open the crates

**Comment by Fresh839** [comment]
Quick question to anyone that reads this. Ike made an annoucement that all mapping changes are stopped for now. Does that mean i should just commit the updated code without the changes i made to each map?

**Comment by Fresh839** [comment]
Ok I think my last commit did some really really strange stuff. Why does it suddenly request a review from Crossed?!!
I just wanted to fix merge conflicts god dammit.

Edit: Ok i royally fucked my pr with that last commit. I have no idea how this happened. Will try to go back to a previous commit, then work from there.

**Comment by Kapu1178** [comment]
this is comedy

**Comment by Kapu1178** [comment]
putting this here as a celebration https://cdn.discordapp.com/attachments/565330972707651584/863146873472417812/unknown.png

**Review by Archanial — DISMISSED** [review]
I wonder if I can trigger actions (I can't)

**Review by Archanial — APPROVED** [review]
Code is good, waiting for map unfreeze

**Review by PowerfulBacon — DISMISSED** [review]

**Review by Archanial — CHANGES_REQUESTED** [review]
Delta is broken

**Comment by Fresh839** [comment]
I won't be able to work on this in the next week or so. Not that we are in a hurry anyways. I hope my last commit fixed deltastation though.

**Review by PowerfulBacon — APPROVED** [review]
This isn't a solution, its just an inconvenience to a problem that will be worked around. But code is fine and it passed vote so :shrug:

**Comment by Fresh839** [comment]
Yes it might not be a solution. But it atleast allows for an IC way of punishing security being equipped with advanced weaponry. After this PR you can blame it on the warden as captain/hos/AI if security is equipped with advanced weapons on a greenshift. I never quite understood why all the guns are supposed to be in the armory but after a couple resarch nodes anyone with basic security access can print better guns than the ones in the armory. Whatever. Its the decision of you guys to merge it in the end. I believe it would be a good change but that is not on me to decide ultimately.

**Comment by qwertyquerty** [comment]
🔑

**Comment by ike709** [comment]
🔑

**Comment by francinum** [comment]
🔑

**Pull request closed without merging**